src/query/interpret/awesome_memgraph_functions.cpp

```diff
@@ -1383,6 +1397,281 @@ TypedValue DateTime(const TypedValue *args, int64_t nargs, const FunctionContext
   return TypedValue(utils::ZonedDateTime(zoned_date_time_parameters), ctx.memory);
 }
 
+TypedValue ToEnum(const TypedValue *args, int64_t nargs, const FunctionContext &ctx) {
+  FType<String, Optional<String>>("toEnum", args, nargs);
+
+  auto const &s1 = args[0].ValueString();
+  if (nargs == 1) {
+    auto enum_val = ctx.db_accessor->GetEnumValue(s1);
+    if (enum_val.HasError()) throw QueryRuntimeException("Invalid enum '{}'", s1);
+    return TypedValue(*enum_val, ctx.memory);
+  }
+  auto const &s2 = args[1].ValueString();
+  auto enum_val = ctx.db_accessor->GetEnumValue(s1, s2);
+  if (enum_val.HasError()) throw QueryRuntimeException("Invalid enum '{}::{}'", s1, s2);
+  return TypedValue(*enum_val, ctx.memory);
+}
+
+TypedValue Point(const TypedValue *args, int64_t nargs, const FunctionContext &ctx) {
```

Comment:
There are multiple combinations in this code and seems like easy to introduce bug. I am not sure for one question below, but what I would rather do is rewrite function that if you start with defining x: you need to also find y: and throw error otherwise. Instead of finding every combination, you are narrowing it down. Seems like it would be less bug prone

Reply:
It is really nice code btw, but seems hard to catch every combination

Reply:
From reading this code I was not able to figure out if we should allow following combination or not:
`RETURN point({longitude:1, latitude:2, srid:7205}).srid;`
This is just my objection

Reply:
7205 we do `Invalid SRID`
7203 we accept, which is a 🐛 should be rejected.